apps/studio/components/interfaces/DataWarehouse/WarehouseAccessTokens.tsx

```diff
@@ -0,0 +1,232 @@
+  DropdownMenuTrigger,
+  Input,
+  cn,
+} from 'ui'
+import { AlertCircle, Link, MoreVertical, TrashIcon } from 'lucide-react'
+import { AlertTitle } from '@ui/components/shadcn/ui/alert'
+import { useParams } from 'common'
+import { useWarehouseAccessTokensQuery } from 'data/analytics/warehouse-access-tokens-query'
+import { GenericSkeletonLoader } from 'ui-patterns'
+import Table from 'components/to-be-cleaned/Table'
+import { useCreateWarehouseAccessToken } from 'data/analytics/warehouse-access-tokens-create-mutation'
+import toast from 'react-hot-toast'
+import { ProjectPausedAlert } from 'components/ui/ProjectPausedAlert'
+import { set } from 'lodash'
+import { useDeleteWarehouseAccessToken } from 'data/analytics/warehouse-access-tokens-delete-mutation'
+import CopyButton from 'components/ui/CopyButton'
+
+const AccessTokenItem = ({
+  token,
+  id,
+  description,
+  onDeleteClick,
+  inserted_at,
+}: {
+  token: string
+  id: string
+  description: string
+  inserted_at: string
+  onDeleteClick: (id: string) => void
+}) => {
+  const formattedInsertedAt = new Date(inserted_at).toLocaleString()
+
+  return (
+    <Table.tr className="group">
+      <Table.td className="max-w-[240px]">{description}</Table.td>
+      <Table.td>{formattedInsertedAt}</Table.td>
+      <Table.td>
+        <div className="flex gap-1 relative">
+          <Input disabled defaultValue={token} type="password" size="tiny" className="flex-grow" />
+          <div className="w-[88px]">
+            <CopyButton
+              type="outline"
+              text={token}
+              className="group-hover:opacity-100 opacity-0 transition-opacity absolute right-0"
+            />
+          </div>
+        </div>
+      </Table.td>
+      <Table.td className="!p-1.5 text-right">
+        <DropdownMenu>
+          <DropdownMenuTrigger className="h-8 w-8 p-2 focus-visible:outline-none">
+            <MoreVertical size="14" />
+          </DropdownMenuTrigger>
+          <DropdownMenuContent className="max-w-40">
+            <DropdownMenuItem
+              className="flex gap-1.5 "
+              onClick={(e) => {
+                e.preventDefault()
+                onDeleteClick(id)
+              }}
+            >
+              <TrashIcon size="14" />
+              Revoke token
+            </DropdownMenuItem>
+          </DropdownMenuContent>
+        </DropdownMenu>
+      </Table.td>
+    </Table.tr>
+  )
+}
+
+const WarehouseAccessTokens = () => {
+  const isProjectActive = useIsProjectActive()
+  const params = useParams()
+  const projectRef = params.ref as string
+
+  const accessTokensQuery = useWarehouseAccessTokensQuery({ projectRef })
+  const hasAccessTokens = accessTokensQuery.isSuccess && accessTokensQuery.data.data.length > 0
+  const [tokenToDelete, setTokenToDelete] = React.useState<string | null>(null)
+  const [showDeleteDialog, setShowDeleteDialog] = React.useState(false)
+
+  const deleteWarehouseAccessToken = useDeleteWarehouseAccessToken({
```

Comment:
just FYI all RQ mutations (refer to the RQ templates) would have a default onError handler, so unless this UI needs something specific, can be left out. Also a good practice to always surface what the error is

Reply:
also for consistency, destructure this

const { mutate: deleteWarehouseAccessToken, isLoading: isDeleting } = useDeleteWarehouseAccessToken()

it looks like this mutation isn't following the same format that the templates have too